dartagnan/src/main/java/com/dat3m/dartagnan/program/Thread.java

```diff
@@ -14,6 +17,9 @@ public Thread(String name, FunctionType funcType, List<String> parameterNames, i
         Preconditions.checkNotNull(entry, "Thread entry event must be not null");
     }
 
+    public Optional<ScopeHierarchy> optScopeHierarchy = Optional.empty();
+    public Optional<HashSet<Thread>> optSyncSet = Optional.empty();
```

Comment:
Made as private.

Comments toward these two optional fields added.

A thread might have several threads it ssw to (e.g. in this [test](https://github.com/KhronosGroup/Vulkan-MemoryModel/blob/master/alloy/tests/ssw0.test#L16-L17)). So I think it should be a HashSet.

The optScopeHierarchy is made as `final`, I think optSyncSet might not be `final` if it's correct to keep it as a `HashSet`?

Reply:
I was asking if it needs to be a HASH set or it can be any set?

You might add threads to The set, but the set instance will always be the same I think so you should be able to use `final`

Reply:
Yeah it doesn't need to be a `HashSet`, changed to `Set` and made as `final` now.